### PR TITLE
Fix uploading artifacts for pull requests with: <\special characters/> in the "title"

### PR DESCRIPTION
**Pull request by cbjeukendrup**
Resolves: #9360

Also bump version of styfle/cancel-workflow-action from 0.5.0 to 0.9.1.

### CODE REVIEW

File: .github/workflows/ci_linux_mu4.yml [hunk 1]
```diff
@@ -79,6 +79,8 @@ jobs:
           DO_PUBLISH='false'
         fi
 
+        ARTIFACT_NAME="$(tr "\":<>|*?\\/" "_" <<<"MU4_${{ github.run_id }}_Lin_${{ github.event.pull_request.number }}_${{ github.event.pull_request.title }}")"
```
Comment (igorkorsukov) [inline .github/workflows/ci_linux_mu4.yml:82]: The build scripts use a variable called `ARTIFACT_NAME`, this is the name of the executable file.
There may be a conflict with this name.
So, I propose to rename, for example, to `UPLOAD_ARTIFACT_NAME`

Comment (cbjeukendrup) [reply .github/workflows/ci_linux_mu4.yml]: Done!

File: .github/workflows/ci_windows_mu4.yml [hunk 2]
```diff
@@ -138,7 +142,7 @@ jobs:
       if: ${{ always() }}
       uses: actions/upload-artifact@v2
       with:
-        name: MU4_${{ github.run_id }}_Win${{ github.event.pull_request.title }}
+        name: ${{ env.UPLOAD_ARTIFACT_NAME }}
```
Comment (Jojo-Schmitz) [inline .github/workflows/ci_windows_mu4.yml:145]: Why not changing this for the 32bit and the portableApp builds too, even with those being commented out at the moment?
Maybe adding a Win32 / PortableApp prefix?

Comment (cbjeukendrup) [reply .github/workflows/ci_windows_mu4.yml:145]: I thought the commented out ones were not really up to date anyway, so initially didn't touch them.
But I updated them too, now.

Comment (Jojo-Schmitz) [reply .github/workflows/ci_windows_mu4.yml:145]: They will get used for Release and probably Beta, maybe Alpha too

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**